src/ipfs/node.ts

```diff
@@ -0,0 +1,400 @@
+/*
+
+(づ￣ ³￣)づ
+
+IPFS node things.
+
+*/
+
+import localforage from "localforage"
+import * as keys from "@libp2p/interfaces/keys"
+import type { IPFS as IPFSCore } from "ipfs-core"
+import type { libp2p } from "ipfs-core/src/components/network"
+import { multiaddr, Multiaddr } from "@multiformats/multiaddr"
+import { IPFSOptions } from "ipfs-core/components/storage"
+import { create as createIpfsNode } from "ipfs-core"
+import { peerIdFromString } from "@libp2p/peer-id"
+
+import * as t from "../common/type-checks.js"
+import { setup } from "../setup/internal.js"
+
+
+// GLOBAL STATE
+
+
+const latestPeerTimeoutIds: { [peer: string]: null | ReturnType<typeof setTimeout> } = {}
+const isSafari = /^((?!chrome|android).)*safari/i.test(globalThis.navigator?.userAgent || "")
+
+
+
+// TYPES
+
+
+type BackOff = {
+  retryNumber: number
+  lastBackoff: number
+  currentBackoff: number
+}
+
+type Status = {
+  connected: boolean
+  lastConnectedAt: number | null
+  latency: number | null
+}
+
+
+type IPFS = IPFSCore & { libp2p: libp2p }
+
+
+
+// OPTIONS
+
+
+/** 🎛️ Connection interval knobs
+ *
+ * KEEP_ALIVE_INTERVAL: Interval to keep the connection alive when online
+ * KEEP_TRYING_INTERVAL: Interval to keep trying the connection when offline
+ * PING_TIMEOUT: How long to wait for a response when pinging an IPFS peer
+ * BACKOFF_INIT: Starting intervals for fibonacci backoff used when establishing a connection
+ */
+const KEEP_ALIVE_INTERVAL =
+  1 * 60 * 1000 // 1 minute
+
+const KEEP_TRYING_INTERVAL =
+  5 * 60 * 1000 // 5 minutes
+
+const PING_TIMEOUT =
```

Comment:
This variable is unused. Was the goal to clean up pings that are taking too long?

Reply:
I tried to use `ipfs.ping` instead of the libp2p ping, which was using this timeout duration. But, that function didn't work for me, so I went back to the libp2p ping which sadly doesn't have a timeout duration option. I'll remove it 👍